File: src/main/java/studio/karllang/karl/errors/SyntaxError/SyntaxError.java
package studio.karllang.karl.errors.SyntaxError;

import studio.karllang.karl.errors.Error;

/** Represents a syntax error in Karl source code. */
public class SyntaxError extends Error {
  public SyntaxError(String message, String filePath, int line, int position) {
    super("Syntax Error", message, filePath, line, position);
  }
}

File: src/main/kotlin/studio/karllang/karl/errors/syntax/SyntaxError.kt
package studio.karllang.karl.errors.syntax

class SyntaxError(
    private val msg: String,
    private val pos: Int,
    private val line: Int,
    private val lineString: String
) : Throwable() {
    fun getMsg(): String {
        return msg
    }

    fun getPosition(): Int {
        return pos
    }

    fun getLine(): Int {
        return line
    }

    fun getLineString(): String {
        return lineString
    }
}